add pytest warning options used for 2023.1

javierggt opened the pull request:
## Description

This PR includes the pytest ignore options that I used for ska3-prime.

We can:
1. merge it as is
2. check that older flags are not needed anymore
3. check whether any of the new flags appear in our code.

## Testing

- [ ] Functional testing

Fixes #

taldcroft commented:
But maybe just merging is the easiest way to find out?

jeanconn commented:
I thought these were "live tested" in that these changes were already used to silence a bunch of the warnings in the outputs that we reviewed for promotion of ska3-prime.

javierggt commented:
I made some changes and force-pushed just now.

I removed the old options and checked that there are no warnings when running testr, so there is no reason to keep them.

I removed an option that I think we need to fix in ska_shell (nose tests being deprecated)

taldcroft commented:
I used this file testing xija and got some warnings still:
```
(ska3) ➜  xija git:(blacken) ✗ pytest xija --pdb
========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.10.8, pytest-7.2.1, pluggy-1.0.0
rootdir: /Users/aldcroft/git, configfile: pytest.ini
plugins: timeout-2.1.0, anyio-3.6.2
collected 43 items                                                                                                                                                       

xija/tests/test_get_model_spec.py .......                                                                                                                          [ 16%]
xija/tests/test_models.py ....................................                                                                                                     [100%]

============================================================================ warnings summary ============================================================================
xija/xija/tests/test_models.py::test_dpa_real[True]
  /Users/aldcroft/miniconda3/envs/ska3/lib/python3.10/site-packages/ipyparallel/client/view.py:8: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
    import imp

xija/xija/tests/test_models.py::test_model_from_chandra_models[acisfp]
  /Users/aldcroft/miniconda3/envs/ska3/lib/python3.10/site-packages/setuptools_scm/git.py:295: UserWarning: git archive did not support describe output
    warnings.warn("git archive did not support describe output")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================== 43 passed, 2 warnings in 51.37s =====================================================================
```
I'm not sure if the one about `git archive` is something we need to understand.

jeanconn reviewed (changes requested):
I was holding my review until seeing more feedback on Tom's latest question about the git archive thing.

javierggt commented:
I used this (or a variation) to produce the results posted in the skare3 PR, where none of these warnings showed up. Everything was green.

Having said that, I just tried it on my mac, using pytest directly, and saw the `imp module` warning. I do not see the other one. Is it because it is in the master version?

I don't have strong feelings. Just tell me what you want to include in this file.

javierggt commented:
There is another warning that shows up, and I have silenced it up to now, but I don't think I should from now on. That's the one about nose tests being deprecated.

jeanconn reviewed (approved):
Might as well merge.

taldcroft reviewed (approved):
Looks good enough.
